### PR TITLE
Implement exit code on errors

### PR DESCRIPTION
**Pull request by angulito**
## Description

Implement exit codes on failures following codes defined on https://github.com/adobe/aio-cli-plugin-cloudmanager/issues/215

⚠️ I have set a special code for deployment warnings, including here the install staged case:
```
Logs:
> The analyser found the following errors for publish :
> [api-regions-exportsimports] com.adobe.aem.temp:com.adobe.aem.developer.console.support:0.1.19.202310190732: Bundle com.adobe.aem.developer.console.support:0.1.19.202310190732 is importing package(s) [Package org.apache.felix.cm.json;version=[1.0,2), Package org.apache.sling.feature;version=[1.8,2), Package com.fasterxml.jackson.dataformat.yaml;version=[2.13,3), Package com.adobe.granite.toggle.api;version=[1.3,1.4)] in start level 20 but no visible bundle is exporting these for that start level in the required version range.
> org.apache.felix.cm.json is exported in regions [com.adobe.aem.internal] but it is imported in regions [com.adobe.aem.deprecated, global]
> com.fasterxml.jackson.dataformat.yaml is exported in regions [com.adobe.aem.internal] but it is imported in regions [com.adobe.aem.deprecated, global]
> org.apache.sling.feature is exported in regions [com.adobe.aem.internal] but it is imported in regions [com.adobe.aem.deprecated, global]
> com.adobe.granite.toggle.api is exported in regions [com.adobe.aem.internal] but it is imported in regions [com.adobe.aem.deprecated, global]
> The analyser found the following errors for author :
> [api-regions-exportsimports] com.adobe.aem.temp:com.adobe.aem.developer.console.support:0.1.19.202310190732: Bundle com.adobe.aem.developer.console.support:0.1.19.202310190732 is importing package(s) [Package org.apache.felix.cm.json;version=[1.0,2), Package org.apache.sling.feature;version=[1.8,2), Package com.fasterxml.jackson.dataformat.yaml;version=[2.13,3), Package com.adobe.granite.toggle.api;version=[1.3,1.4)] in start level 20 but no visible bundle is exporting these for that start level in the required version range.
> org.apache.felix.cm.json is exported in regions [com.adobe.aem.internal] but it is imported in regions [com.adobe.aem.deprecated, global]
> com.fasterxml.jackson.dataformat.yaml is exported in regions [com.adobe.aem.internal] but it is imported in regions [com.adobe.aem.deprecated, global]
> org.apache.sling.feature is exported in regions [com.adobe.aem.internal] but it is imported in regions [com.adobe.aem.deprecated, global]
> com.adobe.granite.toggle.api is exported in regions [com.adobe.aem.internal] but it is imported in regions [com.adobe.aem.deprecated, global]
 ›   Error: [RDECLI:INSTALL_STAGED] Install command was staged, analysers
 ›   detected missing packages that can be fixed with another deployment on top
 ›   of this one. Please check logs. If this is expected, you can ignore exit
 ›   code 40
 ›   Code: INSTALL_STAGED
```

Therefore, you can check after the exit code in the script if the exit code is 40 then it means it is a deployment warning. So if you're expecting this case, you can ignore it. Otherwise, I consider it a failure.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/commands/aem/rde/delete.js [hunk 1]
```diff
@@ -63,13 +68,16 @@ class DeleteCommand extends BaseCommand {
...
+          messageValues: [typeInfo, args.id, serviceInfo],
+        });
       }
     } catch (err) {
       spinner.stop();
-      cli.log(err);
+      if (err instanceof AioError) {
+        throw err;
+      }
+      throw new internalCodes.INTERNAL_DELETE_ERROR({ messageValues: err });
```
Comment (jsedding) [inline src/commands/aem/rde/delete.js]: Would it make sense to factor out the differentiation between `AioError` and internal error? 

```suggestion
handleError(err, internalCodes.INTERNAL_HISTORY_ERROR);
```

or even

```suggestion
internalCodes.INTERNAL_HISTORY_ERROR.handle(err);
```

**Review by jsedding — APPROVED** [review]
This looks really good! I only left one minor suggestion.

**Comment by jsedding** [comment]
For the record, this fixes #43.

**Comment by angulito** [comment]
Also fixes https://github.com/adobe/aio-cli-plugin-aem-rde/issues/18

File: src/commands/aem/rde/delete.js (hunk 1 above)
Comment (angulito) [reply src/commands/aem/rde/delete.js]: Good point - managed at https://github.com/adobe/aio-cli-plugin-aem-rde/pull/53/commits/63f5067cc54d1aa5d8aeb659d7ea3cc01fe6d3b5

**Pull request merged**